core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -182,10 +183,15 @@ trait VecFactory extends SourceInfoDoc {
     implicit sourceInfo: SourceInfo
   ): UInt = {
     val w = (n - 1).bitLength
-    if (n <= 1) 0.U
-    else if (idx.width.known && idx.width.get <= w) idx
-    else if (idx.width.known) idx(w - 1, 0)
-    else (idx | 0.U(w.W))(w - 1, 0)
+    val name = "_vec_index_truncate"
+    // Give any truncation logic a name to make debugging easier
+    autoNameRecursively(name)(prefix(name) {
+      if (n <= 1) WireInit(0.U) // Need the Wire otherwise we emit vec[0] which is illegal FIRRTL.
+      // Other cases do not need a Wire because the literal is truncated to fit.
+      else if (idx.width.known && idx.width.get <= w) idx
+      else if (idx.width.known) idx(w - 1, 0)
+      else (idx | 0.U(w.W))(w - 1, 0)
+    })
```

Comment:
Nit: A match would be more readable, IMO.

Reply:
Also, the naming change would be better broken out. However, I'll allow it.

Reply:
> Nit: A match would be more readable, IMO.

Agreed, although I'm mostly keeping the logic untouched.

> Also, the naming change would be better broken out. However, I'll allow it.

That's fair, I think I'll back it out and then the diff will better justify my decision to not change it into a match 😇